### PR TITLE
[IE11] multi-select onBlur bug

### LINKED ISSUE #596: [fullstory] District filter remains open after interacting with other parts of the page
**What needs to be done? Why does it need to be done?**
From viewing a Fullstory session, it appears that one users's district filter remains open even after they click on the case type filter and interact with other parts of the page. I did not see this behavior occur for any other users. The user was using IE11.

**Additional context**
Fullstory is sometimes a little janky, so I'm not sure if this was actually occurring, or the video was off. Can close if this ends up being not reproducible.

### PR DESCRIPTION
**Pull request by jovergaag**
## Description of the change

We had a bug in IE11 where clicking outside of the multi-select menu did not close the menu. This bug only existed react-scripts builds and not in webpack builds which was strange.

The bug was fixed by cleaning up how we are searching for the select input to pass to the ValueContainer. We were not finding the input correctly with the react-scripts build (timing maybe) so the ref was lost, and therefore the focus/blur was broken.

This fix ended up breaking the `closeMenuOnSelect={false}` for IE11, so I manually changed the `onFocus` event handler to help IE11 out.

This has been deployed to staging for QA vs what you see in production.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #596

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/components/controls/MultiSelect/ValueContainer.js [hunk 1]
```diff
@@ -24,10 +24,9 @@ import { optionPropType } from "../../propTypes";
 const ValueContainer = ({ allOptions, summingOption, children, ...props }) => {
   const { selectProps, getValue } = props;
   const values = getValue();
-
-  const selectInput = React.Children.toArray(children).find((input) =>
-    ["DummyInput", "Input"].includes(input.type.name)
-  );
+  const selectInput = React.Children.map(children, (child) => {
+    return child.type === components.Input ? child : null;
```
Comment (macfarlandian) [inline src/components/controls/MultiSelect/ValueContainer.js:28]: what was "DummyInput", just some legacy thing we don't need to support anymore?

**Review by macfarlandian — COMMENTED** [review]
I also did a quick QA on IE 11 in browserstack and I am now truly and deeply feeling your pain

File: src/components/controls/MultiSelect/ValueContainer.js (hunk 1 above)
Comment (jovergaag) [reply src/components/controls/MultiSelect/ValueContainer.js:28]: I *think the "DummyInput" was being used to try to focus the select input at some point but ended up going a different route. I found a blog post about creating a custom `ValueContainer` and it looks like this section was copied directly from that blog. A while back I found a line of code that was commented out that was trying to use the "DummyInput" to focus the input, so it looks like someone was struggling with this focus problem a while ago, but ended up not using the "DummyInput" in the end.

**Comment by jovergaag** [comment]
> I also did a quick QA on IE 11 in browserstack and I am now truly and deeply feeling your pain

Thanks. It is truly a test of patience. But already much better than it was 3 weeks ago!

**Review by macfarlandian — APPROVED** [review]

**Comment by macfarlandian** [comment]
one of these days I will have to get a VM testing environment running for IE 11. today is not that day

**Pull request merged**